src/install.c

```diff
@@ -98,19 +98,33 @@ gboolean determine_slot_states(GError **error)
 	mountlist = g_unix_mounts_get(NULL);
 	for (GList *l = mountlist; l != NULL; l = l->next) {
 		GUnixMountEntry *m = (GUnixMountEntry*)l->data;
-		g_autofree gchar *devicepath = resolve_loop_device(g_unix_mount_get_device_path(m));
-		RaucSlot *s = find_config_slot_by_device(r_context()->config,
-				devicepath);
-		if (s) {
-			/* We might have multiple mount entries matching the same device and thus the same slot.
-			 * To avoid leaking the string returned by g_unix_mount_get_mount_path() here, we skip all further matches
-			 */
-			if (s->ext_mount_point) {
-				break;
-			}
-			s->ext_mount_point = g_strdup(g_unix_mount_get_mount_path(m));
-			g_debug("Found external mountpoint for slot %s at %s", s->name, s->ext_mount_point);
+		g_autofree gchar *devicepath = NULL;
+		RaucSlot *s = NULL;
+
+		/* Relative paths will be from something like tmpfs, cgroups,
+		 * etc. which we want to ignore rather than try to follow some
+		 * random symlink named "tmpfs" in the CWD.
+		 */
+		if (g_unix_mount_get_device_path(m)[0] != '/')
```

Comment:
I'm not sure this is correct for ubifs, which has device names such as `ubi0:data`.